src/caret_analyze/plot/timeseries/frequency_timeseries.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2021 Research Institute of Systems Planning, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional, Sequence, Tuple, Union
+
+import pandas as pd
+
+from ..metrics_base import MetricsBase
+from ...exceptions import InvalidArgumentError
+from ...record import Frequency, RecordsInterface
+from ...runtime import CallbackBase, Communication, Publisher, Subscription
+
+TimeSeriesTypes = Union[CallbackBase, Communication, Union[Publisher, Subscription]]
+
+
+class FrequencyTimeSeries(MetricsBase):
+    """Class that provides frequency timeseries data."""
+
+    def __init__(
+        self,
+        target_objects: Sequence[TimeSeriesTypes]
+    ) -> None:
+        super().__init__(target_objects)
+
+    def to_dataframe(self, xaxis_type: str = 'system_time') -> pd.DataFrame:
+        """
+        Get frequency timeseries data for each object in pandas DataFrame format.
+
+        Parameters
+        ----------
+        xaxis_type : str
+            Type of time for timestamp.
+            "system_time", "index", or "sim_time" can be specified.
+            The default is "system_time".
+
+        Returns
+        -------
+        pd.DataFrame
+            Multi-column frequency DataFrame.
+
+        Notes
+        -----
+        xaxis_type "system_time" and "index" return the same DataFrame.
+
+        """
+        timeseries_records_list = self.to_timeseries_records_list()
+        if xaxis_type == 'sim_time':
+            self._convert_timeseries_records_to_sim_time(timeseries_records_list)
```

Comment:
These lines seem duplicated.
Same implementations are located in to_timeseries_records_list
```suggestion
        timeseries_records_list = self.to_timeseries_records_list(xaxis_type)
```

Reply:
Fixed in: https://github.com/tier4/CARET_analyze/pull/170/commits/d1f97e98032f433cbe4f2a52748f66b4d26a21b8